### PR TITLE
ref(metrics): Render info when creating new span attribute

### LINKED ISSUE #75124: Poor man's autocomplete
Image
We should tell users here that if you've created an attribute with setAttribute/setData and don't see it in autocomplete, it's ok, they can just click enter.

Notion example:
Image

Vjeran:
Note next to autocomplete missing which would say: “There might be more things…”

### PR DESCRIPTION
**Pull request by priscilawebdev**
### Goal

- In the metrics modal, an extra message shall be displayed above the 'Create "new-span-attribute"' option in the dropdown, with a link to the docs for instrumenting custom metrics

### Notes
- This PR also updates the blue alert copy shown when the form changes
- This PR changes 'Custom: new-value' to 'Create new-value'

**Preview**

https://github.com/user-attachments/assets/dcb9c94d-4dc1-4c36-90a9-e194155aabc0

closes https://github.com/getsentry/sentry/issues/75124

### CODE REVIEW

**Comment by vgrozdanic** [comment]
image

When selecting group and filter by, should we also show the message about creating new tag if it isn't already present there (which is very likely scenario)?

**Comment by priscilawebdev** [comment]
hey @shellmayr yes I saw that 🙂 I did not want to give the same left padding to the new text, because it would look even more strange and I did not remove it from the Create because it is like that everywhere. Let me discuss this internally and work on it in a follow-up.

**Comment by priscilawebdev** [comment]
@vgrozdanic is that is a good idea and I can create an extra issue for that. Do you know if we already have a doc describing it?

File: static/app/views/settings/projectMetrics/metricsExtractionRuleForm.tsx
Comment (vgrozdanic) [inline static/app/views/settings/projectMetrics/metricsExtractionRuleForm.tsx:356]: I don't think we should be using this link, or maybe the docs should be updated, because currently old `metrics` API is mentioned on those docs, and we are trying to move away from it. There is no mention here about new span based attributes, or attaching an attribute to a span

**Comment by Jesse-Box** [comment]
@priscilawebdev one option would be to add a plus icon to the left of the “Create...”. My thinking is that initially the users mental model might be to “select something from this list,” but eventually, they're “Adding something”...

Let me know if that jives with you or not.

File: static/app/views/settings/projectMetrics/metricsExtractionRuleForm.tsx
Comment (priscilawebdev) [reply static/app/views/settings/projectMetrics/metricsExtractionRuleForm.tsx:356]: Afaik @ale-cota is working on updating this doc

**Comment by vgrozdanic** [comment]
> @vgrozdanic that is a good idea and I can create an extra issue. Do you know if we already have a doc describing it?

I don't know if we have a doc describing it, maybe @ale-cota has more details.

But I think that can be done as a part of this PR, because the same problem is happening with both "Filter and group by" and "Span attribute" fields, and that is the creation of something that we don't have indexed yet, and it shouldn't feel weird to create something that user knows should be there

**Comment by priscilawebdev** [comment]
> > @vgrozdanic that is a good idea and I can create an extra issue. Do you know if we already have a doc describing it?
> 
> I don't know if we have a doc describing it, maybe @ale-cota has more details.
> 
> But I think that can be done as a part of this PR, because the same problem is happening with both "Filter and group by" and "Span attribute" fields, and that is the creation of something that we don't have indexed yet, and it shouldn't feel weird to create something that user knows should be there

All right! I can introduce this change as soon as we clarify the link to the docs 😉

File: static/app/views/settings/projectMetrics/metricsExtractionRuleForm.tsx
Comment (vgrozdanic) [reply static/app/views/settings/projectMetrics/metricsExtractionRuleForm.tsx:356]: Nice, ok, got it. Should we wait with this PR until the docs are updated, maybe we can link the specific section of those docs?

Comment (priscilawebdev) [reply static/app/views/settings/projectMetrics/metricsExtractionRuleForm.tsx:356]: yes, I think we should but I will leave it up to @ale-cota  🙂

Comment (vgrozdanic) [reply static/app/views/settings/projectMetrics/metricsExtractionRuleForm.tsx:356]: Perfect, thank you :)

Comment (ale-cota) [reply static/app/views/settings/projectMetrics/metricsExtractionRuleForm.tsx:356]: We don't have to wait for the docs update to merge this, since it's behind a feature flag and only available to Sentry internal. Thanks!

**Review by ale-cota — APPROVED** [review]

File: static/app/views/settings/projectMetrics/metricsExtractionRuleForm.tsx
Comment (priscilawebdev) [inline static/app/views/settings/projectMetrics/metricsExtractionRuleForm.tsx:463]: @vgrozdanic I have introduced the changes you suggested 😉

File: static/app/components/forms/controls/selectOption.tsx
Comment (priscilawebdev) [inline static/app/components/forms/controls/selectOption.tsx:63]: @Jesse-Box I added your suggestion here, but since I updated a core component, I would like to wait for @vuluongj20 opinion as well. 

it looks like this 
image

File: static/app/components/forms/controls/selectOption.tsx
Comment (priscilawebdev) [inline static/app/components/forms/controls/selectOption.tsx:78]: I've implemented this change because the values looked odd when no item was selected—they were shifted to the right. @vuluongj20, could you please review and let me know if this works for you?

**Before:**
image


**After:**

image
image

Comment (priscilawebdev) [reply static/app/components/forms/controls/selectOption.tsx:78]: ^ @shellmayr

File: static/app/views/settings/projectMetrics/metricsExtractionRuleForm.tsx
Comment (obostjancic) [inline static/app/views/settings/projectMetrics/metricsExtractionRuleForm.tsx:367]: I checked this out and we probably don't need this prop anymore since the default is `Create [value]` (same for other occurences)

File: static/app/views/settings/projectMetrics/metricsExtractionRuleForm.tsx
Comment (obostjancic) [inline static/app/views/settings/projectMetrics/metricsExtractionRuleForm.tsx:311]: This will return false for custom span attributes that are already present in the dropdown so i would propose renaming it to:
```suggestion
  const isNewCustomSpanAttribute = useCallback((value: string) => {
```

File: static/app/components/forms/controls/selectOption.tsx
Comment (shellmayr) [reply static/app/components/forms/controls/selectOption.tsx:78]: Looks much better now 🙏

**Review by obostjancic — APPROVED** [review]

File: static/app/views/settings/projectMetrics/metricsExtractionRuleForm.tsx
Comment (priscilawebdev) [reply static/app/views/settings/projectMetrics/metricsExtractionRuleForm.tsx:367]: the new version has double quotes, but I don't have a strong opinion on this.

File: static/app/views/settings/projectMetrics/metricsExtractionRuleForm.tsx
Comment (vgrozdanic) [inline static/app/views/settings/projectMetrics/metricsExtractionRuleForm.tsx:621]: Is there a reason why this has to be done like this?

File: static/app/components/forms/controls/selectOption.tsx
Comment (priscilawebdev) [inline static/app/components/forms/controls/selectOption.tsx:59]: I’m not fond of using "__isNew__", as it doesn’t feel quite right, but I don’t have a better alternative at the moment. For what it’s worth, newer versions of react-select still use it. You can see an example here: [react-select GitHub](https://github.com/JedWatson/react-select/blob/06e34882638d1526b9f5a1238bb567a3e9460ce5/packages/react-select/src/useCreatable.ts#L98-L102).

Comment (priscilawebdev) [reply static/app/components/forms/controls/selectOption.tsx:59]: We could create our own prop to check if a value is new, but this would require manually updating every place we use it. Otherwise, the "create" option won’t show the plus icon. That might be fine as well

Comment (vgrozdanic) [reply static/app/components/forms/controls/selectOption.tsx:59]: I would rather not show plus icon, than using the `__isNew__` attribute over which we have no control, and is not a public property, but I have no strong opinion here

File: static/app/views/settings/projectMetrics/metricsExtractionRuleForm.tsx
Comment (priscilawebdev) [reply static/app/views/settings/projectMetrics/metricsExtractionRuleForm.tsx:621]: I could use `${space(2)}`, but I prefer to be more explicit about the value. However, it might not be clear enough

Screenshot 2024-08-01 at 09 34 30
Screenshot 2024-08-01 at 09 34 37

Comment (priscilawebdev) [reply static/app/views/settings/projectMetrics/metricsExtractionRuleForm.tsx:621]: the options are wrapped with elements that have different paddings. 1 element has `${space(0.5)} ` and the other `${space(1.5)}`

Comment (vgrozdanic) [reply static/app/views/settings/projectMetrics/metricsExtractionRuleForm.tsx:621]: Ok, got it, makes sense then

File: static/app/components/forms/controls/selectOption.tsx
Comment (priscilawebdev) [reply static/app/components/forms/controls/selectOption.tsx:59]: the whole data prop does not have a type, but we are using props from it 🙂 the `__isNew__` feels odd only because of the `__`

**Review by vgrozdanic — APPROVED** [review]

File: static/app/components/forms/controls/selectOption.tsx
Comment (priscilawebdev) [reply static/app/components/forms/controls/selectOption.tsx]: I removed this change from this PR and created a [new one](https://github.com/getsentry/sentry/pull/75434) 😉

**Pull request merged**
